src/components/panel/panel.spec.js

```diff
@@ -643,6 +606,226 @@ describe('$mdPanel', function() {
       expect(scrollMaskEl).not.toExist();
       expect($mdUtil.disableScrollAround).toHaveBeenCalled();
     });
+
+    describe('animation hooks: ', function() {
+      it('should call onDomAdded if provided when adding the panel to the DOM',
+          function() {
+            var onDomAdded = jasmine.createSpy();
+            var config = angular.extend(
+                {'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+            panelRef = $mdPanel.create(config);
+            panelRef.attach();
+
+            flushPanel();
+
+            expect(onDomAdded).toHaveBeenCalled();
+            expect(PANEL_EL).toExist();
+            expect(PANEL_WRAPPER_CLASS).toHaveClass(HIDDEN_CLASS);
+          });
+
+      it('when onDomAdded resolves continue resolving the ' +
+          'rest of promise chain', function() {
+            var attachResolved = false;
+            var onDomAddedCalled = false;
+
+            var onDomAdded = function() {
+              onDomAddedCalled = true;
+              return $q.when(this);
+            };
+
+            var config = angular.extend(
+                {'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+            expect(PANEL_EL).not.toExist();
+
+            panelRef = $mdPanel.create(config);
+
+            panelRef.open().then(
+                function() {
+                  attachResolved = true;
+                });
+            flushPanel();
+
+            expect(onDomAddedCalled).toBe(true);
+            expect(PANEL_EL).toExist();
+            expect(attachResolved).toBe(true);
+            expect(panelRef.isAttached).toEqual(true);
+            expect(panelRef._panelContainer).not.toHaveClass(HIDDEN_CLASS);
+          });
+
+      it('when onDomAdded rejects, open should also reject', function() {
+        var openRejected = false;
+
+        var onDomAdded = function() {
+          return $q.reject();
```

Comment:
Stick with the same structure as above. In both the above tests, you also verify that onDomAdded is called.